test/poll_for_state.go

```diff
@@ -109,9 +117,43 @@ func (p blockPoller) findTimeout(ctx context.Context, height uint64, packet ibc.
 		return zero, err
 	}
 	for _, t := range timeouts {
+		p.pollErr.PushSearched(t)
 		if t.Packet.Equal(packet) {
 			return t, nil
 		}
 	}
 	return zero, ErrNotFound
 }
+
+type pollError struct {
+	error
```

Comment:
If I'm following this correctly, we embed `error` here so that the `.Error()` method returns `p.error.Error()`; the only special case of presentation is the `+v` verb, which emits the expanded form of target packet and searched; and it looks like `+v` is used internally by `require` when formatting errors.

Reply:
> it looks like +v is used internally by require when formatting errors.

Yeah, that could be brittle if testify changes its behavior. But I thought it was nice to let the caller decide if it wants a short or very long error. 

Fwiw, I often embed with private types for convenience, especially if it's an adapter. But I almost never embed with public/exported types. But happy to never embed if we decide that's our code style.